kedro/framework/project/__init__.py

```diff
@@ -39,6 +41,25 @@ def validate(self, settings, *args, **kwargs):
                 )
 
 
+class _HasSharedParentClassValidator(Validator):
+    """A validator to check that the parent of the default class is an ancestor of
+    the settings value."""
+
+    def validate(self, settings, *args, **kwargs):
+        super().validate(settings, *args, **kwargs)
+
+        default_class = self.default(settings, self)
```

Comment:
so default_class here would be either `ConfigLoader` if not specified or `TemplatedConfigLoader`, correct?

Reply:
No, `default_class` comes from `default=_get_default_class("kedro.config.ConfigLoader")` so is always `ConfigLoader`. The actual specified value is given in `setting_value = getattr(settings, name)`.

These dynaconf validators are very confusing to read I know but we're following their standard syntax here 😅